File: app/src/main/java/com/example/heroesofiu3/Java/GameField.java
/*
package com.example.heroesofiu3;

import java.util.ArrayList;
import java.util.List;

public class GameField {
    private final Integer width;
    private final Integer height;
    private final List<List<Cell>> cells;

    public GameField(Integer width, Integer height) {
        this.width = width;
        this.height = height;
        this.cells = new ArrayList<>();

        for (int x = 0; x < width; x++){
            List<Cell> column = new ArrayList<>();
            for (int y = 0; y < height; y++){
                column.add(new Cell(x,y));
            }
            cells.add(column);
        }
    }

    public Integer getWidth() {
        return width;
    }

    public Integer getHeight() {
        return height;
    }

    public Cell getCell(int x, int y) {
        if(x >= 0 && x <= width && y >= 0 && y <= height){
            return cells.get(x).get(y);
        }
        return null;
    }
}
*/

File: app/src/main/java/com/example/heroesofiu3/domain/entities/gameField/GameField.kt
package com.example.heroesofiu3.domain.entities.gameField

import com.example.heroesofiu3.domain.game.initializeField

class GameField(val width: Int, val height: Int) {

    internal val cells: List<List<Cell>> = List(width) { x ->
        List(height) { y ->
            Cell(
                x, y,
                terrain = if (x == y) Terrain.ROAD
                else if (x <= 4 && y <= 4) Terrain.FRIENDLY
                else if (x >= 5 && y >= 5) Terrain.ENEMY
                else Terrain.OBSTACLE
            )
        }
    }

    fun getCell(x: Int, y: Int): Cell? {
        return cells.getOrNull(x)?.getOrNull(y)
    }

    fun getCellList(): List<Cell> {
        val cellList = mutableListOf<Cell>()
        for(list in cells){
            for (cell in list){
                cellList.add(cell)
            }
        }
        return cellList
    }

    fun reset() {
        for (row in cells) {
            for (cell in row) {
                cell.unit = null
                cell.castle = null
                val x = cell.x
                val y = cell.y
                cell.terrain = if (x == y) Terrain.ROAD
                else if (x <= 4 && y <= 4) Terrain.FRIENDLY
                else if (x >= 5 && y >= 5) Terrain.ENEMY
                else Terrain.OBSTACLE
            }
        }
        initializeField(this)
    }
}